holoviews/tests/core/data/test_ibisinterface.py

```diff
@@ -303,3 +320,90 @@ def test_dataset_iloc_ellipsis_list_cols(self):
 
         def test_dataset_boolean_index(self):
             raise SkipTest("Not supported")
+
+    def test_range(self):
+        assert IbisInterface.range(self.reference_table, "date") == (np.datetime64('2022-01-03'), np.datetime64('2022-01-09'))
+        assert IbisInterface.range(self.reference_table, "string") == ('Mon', 'Sun')
+        assert IbisInterface.range(self.reference_table, "numerical") == (np.float64(1.1), np.float64(5.5))
+
+    def test_dimension_type(self):
+        assert IbisInterface.dimension_type(self.reference_table, "date") is np.datetime64
+        assert IbisInterface.dimension_type(self.reference_table, "string") is np.object_
+        assert IbisInterface.dimension_type(self.reference_table, "numerical") is np.float64
+
+    def test_datetime_xaxis(self):
+        """Test to make sure a DateTimeAxis can be identified for the bokeh backend"""
+        plot_ibis = Curve(self.reference_table, kdims="date", vdims="actual")
+        # When
+        plot_bokeh = render(plot_ibis, "bokeh")
+        xaxis, yaxis = plot_bokeh.axis
+        # Then
+        assert isinstance(xaxis, bokeh_axes.DatetimeAxis)
+        assert isinstance(yaxis, bokeh_axes.LinearAxis)
+
+    def test_categorical_xaxis(self):
+        """Test to make sure a Categorical axis can be identified for the bokeh backend"""
+        plot_ibis = Curve(self.reference_table, kdims="string", vdims="actual")
+        # When
+        plot_bokeh = render(plot_ibis, "bokeh")
+        xaxis, yaxis = plot_bokeh.axis
+        # Then
+        assert isinstance(xaxis, bokeh_axes.CategoricalAxis)
+        assert isinstance(yaxis, bokeh_axes.LinearAxis)
+
+    def test_numerical_xaxis(self):
+        """Test to make sure a LinearAxis axis can be identified for the bokeh backend"""
+        plot_ibis = Curve(self.reference_table, kdims="numerical", vdims="actual")
+        # When
+        plot_bokeh = render(plot_ibis, "bokeh")
+        xaxis, yaxis = plot_bokeh.axis
+        # Then
+        assert isinstance(xaxis, bokeh_axes.LinearAxis)
+        assert isinstance(yaxis, bokeh_axes.LinearAxis)
```

Comment:
First of all, really appreciate your work investigating these issues. That said these tests can't live here. The plotting and data parts of HoloViews are separated for a reason and if test coverage of the data parts is sufficiently extensive then there's no reason to ever test the plotting code with different data backends.

Reply:
>if test coverage of the data parts is sufficiently extensive

To be clear, that is of course clearly not the case since you're discovering these issues.

Reply:
Besides the non-comprehensive test, the missing docstrings and type annotations from the base `Interface` also makes it harder than it has to be, to know what a method on the interface actually should do and return.

Reply:
How would someone be able to setup appropriate tests for a backend @philippjfr? If I look into the `tests/core/data` folder I see nothing that systematically tests the same things across backends?

And knowing what to test for is also quite difficult due to missing docstrings, type annotations, very complex functions and similar.

Reply:
There are various baseclasses in tests/core/data which implement systematic tests which are then subclassed for each interface.

Reply:
The best thing to do is probably to continue extending `tests/core/data/base` `HeterogeneousColumnTests` with more tests.

Reply:
Specifically testing for datetime columns seems to be pretty lacking.